netbox/utilities/views.py

```diff
@@ -140,6 +144,51 @@ def get_return_url(self, request, obj=None):
         return reverse('home')
 
 
+class GetRelatedModelsMixin:
+    """
+    Provides logic for collecting all related models for the currently viewed model.
+    """
+
+    def get_extra_related_models(self, request, instance):
```

Comment:
What is the purpose of this method? It seems like maybe a hook for monkey patching, which IMO should be discouraged. If needed, we could potentially extend the underlying `get_related_models()` utility function (not the class method) to collect additional types of relationships.

Or, if it's intended to be overridden naturally, we could introduce an `extra` keyword argument to the `get_related_models()` method to pass these.

Reply:
You are right, its intended to be used as a hook by view classes, e.g. when not only direct relations should be collected.

Although I prefer a generic approach that implements some logic in the utility function, I don't know how we should determine which transient relations are interesting for object views. So for now, I'd remove this method and add an `extra` keyword argument to the class method instead.